### PR TITLE
Floater position generators

### PR DESCRIPTION
**Pull request by anirban89**
made some changes in the genrators.py to include input configurations for floater (2d or 3d)

### CODE REVIEW

File: floater/generators.py [hunk 1]
```diff
@@ -97,13 +97,34 @@ def parcel_area(self, latlon=True):
...
+         - should profile ( > 0 = return cycle (in s) to surface)
+         - remain at depth ( = 0 )
+         - is a 3D float ( = -1 )
+         - should be advected WITHOUT additional noise (= -2 ); 
+	this implies that the float is non-profiling
+         - is a mooring ( = -3 ); i.e. the float is not advected
+	mesh : choice of mesh
+	 - 'rect' : rectangular cartesian
+	 - 'hex' : hexagonal
+	"""
```
Comment (rabernat) [inline floater/generators.py]: Love the docstring! excellent

Comment (anirban89) [inline floater/generators.py]: what would the test involve exactly?

Comment (rabernat) [inline floater/generators.py]: You can look at the existing tests:
https://github.com/rabernat/floater/blob/master/floater/test/test_generators.py

The simplest thing to do is just call the function and make sure it doesn't error.

A more thorough test would make sure the contents of the generated file are correct.

**Comment by rabernat** [comment]
@anirban89 could you please finish this PR.

File: floater/generators.py [hunk 2]
```diff
@@ -97,13 +97,34 @@ def parcel_area(self, latlon=True):
...
-    def to_mitgcm_format(self, filename, tstart=0, mesh='rect'):
-    	#xx, yy = np.meshgrid(x, y)
+    def to_mitgcm_format(self, filename, tstart=0, iup=0, mesh='rect'):
+	"""Output floatset in MITgcm format
+	PARAMETERS
+	----------
+	filename : The filename to save the floatset data in 
+		   (e.g.float.ini.pos.hex.bin)
+	tstart : time for float initialisation (default = 0)
+	iup : flag if the float
```
Comment (anirban89) [inline floater/generators.py]: use numpydoc syntax for discrete options:
https://github.com/numpy/numpy/blob/master/doc/HOWTO_DOCUMENT.rst.txt

**Pull request merged**